### PR TITLE
Improve `run_federation_tests.py` and add `xunit-viewer` image

### PR DESCRIPTION
**Pull request by alanking**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -202,3 +202,26 @@ ssh-add -k <path to private keys>
...
+docker run --rm -d \
+    -v /path/to/test-results/logs:/results:ro \
+    -p 3000:3000 \
+    xunit-viewer -r /results -s
+```
+
+This does the following:
+ 1. Runs the xunit-viewer container with the specified test results as a server on the default port.
+ 2. Provides `/results` as a volume mount in the container. `/path/to/test-results` is the location of the test results as specified by the `--output-directory`/`-o` option for the test running scripts.
+ 3. Exposes port 3000 in the container as 3000 on the host. This is hte default port for the `xunit-viewer` server.
```
Comment (trel) [inline README.md:226]: "hte"- > "the"

Comment (alanking) [reply README.md]: Done

**Comment by alanking** [comment]
I've been using this for a few weeks now and it seems to be working well. Added some other changes regarding generating negotiation keys. Taking out of draft.

File: run_federation_tests.py [hunk 2]
```diff
@@ -151,9 +154,28 @@
...
-        logging.warning('collecting logs [{}]'.format(output_directory))
-        logs.collect_logs(ctx.docker_client, ctx.irods_containers(), output_directory)
+        if args.save_logs:
+            try:
+                logging.error('collecting logs [{}]'.format(output_directory))
+
+                # collect the usual logs
+                logs.collect_logs(ctx.docker_client, ctx.irods_containers(), output_directory)
+
+                # TODO: This produced errors... why do the federation tests show up somewhere else?
```
Comment (trel) [inline run_federation_tests.py:164]: where do they appear?

**Comment by trel** [comment]
yes, rest looks good upon inspection.   elegant truncation to 32 on the negotiation keys.

**Comment by korydraughn** [comment]
Same here. Any left to do?

File: run_federation_tests.py (hunk 2 above)
Comment (alanking) [reply run_federation_tests.py]: The logs are only appearing in the container that actually ran the test. Adjusted.

**Comment by alanking** [comment]
I think it's good to go now. I'll go ahead and squash and # and merge if there are no objections

**Comment by trel** [comment]
squash and #.

**Comment by trel** [comment]
and merge, sorry.    go go go.

**Pull request merged**